### DISCUSSION TITLE
Function List

### DISCUSSION THREAD

**Discussion by hnimminh** [tayler6000/pyVoIP#87] [Q&A] [answered]
Hi @tayler6000, 
Thanks for your work on [pyVoIP](https://github.com/tayler6000/pyVoIP), Can you help to advise:
* Is pyVoIP support TCP transport?
* Will you plan to support peer to peer out call without REGISTER require? (Register require is not valid reason to block outcall)
* Is there any way to provide the function of custom SIP invite request?, eg: i likely to modify callid, user-agent, from-user, or add extra SIP header (X-ABC-HEADER,..)

Thanks for your time

**Comment by tayler6000** [comment, marked as answer]
Hello!

pyVoIP will support the first two items in 2.0, however I didn't have plans to implement the last one, though that is something that could be done.

**Reply by ConnorEhrgood** [reply]
Hello! Has peer to peer without registration been added yet?
